"git observe" panics when given a non-existing branch name

### Steps to reproduce

run `git observe zonk` when there is no branch "zonk" locally and remotely

### Command output

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x28 pc=0x1032a9eb4]

goroutine 1 [running]:
github.com/git-town/git-town/v14/src/cmd.determineObserveData({_, _, _}, {{0x10364b8f8, 0x1400019ad38}, {0x14000195190}, {0x1400025d230, 0x1400025d350}, {0x1400019ac78}, {0x103648f78, ...}, ...})
        /Users/kevlar/d/git-town/src/cmd/observe.go:145 +0x474
github.com/git-town/git-town/v14/src/cmd.executeObserve({0x14000196b00, 0x1, 0x1}, 0x0)
        /Users/kevlar/d/git-town/src/cmd/observe.go:65 +0xf4
github.com/git-town/git-town/v14/src/cmd.observeCmd.func1(0x1400027e900?, {0x14000196b00, 0x1, 0x1})
        /Users/kevlar/d/git-town/src/cmd/observe.go:46 +0x44
github.com/spf13/cobra.(*Command).execute(0x14000269508, {0x14000196ac0, 0x1, 0x1})
        /Users/kevlar/d/git-town/vendor/github.com/spf13/cobra/command.go:983 +0x840
github.com/spf13/cobra.(*Command).ExecuteC(0x14000248308)
        /Users/kevlar/d/git-town/vendor/github.com/spf13/cobra/command.go:1115 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/kevlar/d/git-town/vendor/github.com/spf13/cobra/command.go:1039
github.com/git-town/git-town/v14/src/cmd.Execute()
        /Users/kevlar/d/git-town/src/cmd/core.go:38 +0x3a8
main.main()
        /Users/kevlar/d/git-town/main.go:18 +0x28
```


### Additional information

This should print an error like "there is no branch zonk to observe".

--- a/features/contribute/edge_cases/non_existing_branch.feature
+++ b/features/contribute/edge_cases/non_existing_branch.feature
@@ -1,9 +1,7 @@
 Feature: cannot make non-existing branches contribution branches
 
   Background:
-    Given the current branch is a feature branch "feature"
-    And an uncommitted file
-    When I run "git-town contribute feature non-existing"
+    When I run "git-town contribute non-existing"
 
   Scenario: result
     Then it runs no commands
@@ -11,12 +9,11 @@ Feature: cannot make non-existing branches contribution branches
       """
       there is no branch "non-existing"
       """
-    And the current branch is still "feature"
-    And the uncommitted file still exists
+    And the current branch is still "main"
     And there are still no contribution branches
 
   Scenario: undo
     When I run "git-town undo"
     Then it runs no commands
     And there are still no contribution branches
-    And the current branch is still "feature"
+    And the current branch is still "main"

--- a/features/observe/edge_cases/non_existing_branch.feature
+++ b/features/observe/edge_cases/non_existing_branch.feature
@@ -1,10 +1,9 @@
 Feature: cannot observe non-existing branches
 
   Background:
-    Given the current branch is a feature branch "feature"
-    And an uncommitted file
-    When I run "git-town observe feature non-existing"
+    When I run "git-town observe non-existing"
 
+  @this
   Scenario: result
     Then it runs no commands
     And it prints the error:

--- a/features/observe/edge_cases/non_existing_branch.feature
+++ b/features/observe/edge_cases/non_existing_branch.feature
@@ -3,19 +3,17 @@ Feature: cannot observe non-existing branches
   Background:
     When I run "git-town observe non-existing"
 
-  @this
   Scenario: result
     Then it runs no commands
     And it prints the error:
       """
-      there is no branch "non-existing"
+      remote branch "origin/non-existing" doesn't seem to exist
       """
-    And the current branch is still "feature"
-    And the uncommitted file still exists
+    And the current branch is still "main"
     And there are still no observed branches
 
   Scenario: undo
     When I run "git-town undo"
     Then it runs no commands
     And there are still no observed branches
-    And the current branch is still "feature"
+    And the current branch is still "main"

--- a/features/observe/edge_cases/non_existing_branch.feature
+++ b/features/observe/edge_cases/non_existing_branch.feature
@@ -7,7 +7,7 @@ Feature: cannot observe non-existing branches
     Then it runs no commands
     And it prints the error:
       """
-      remote branch "origin/non-existing" doesn't seem to exist
+      there is no branch "non-existing"
       """
     And the current branch is still "main"
     And there are still no observed branches

--- a/src/cmd/contribute.go
+++ b/src/cmd/contribute.go
@@ -146,7 +146,7 @@ func determineContributeData(args []string, repo execute.OpenRepoResult) (contri
 		trackingBranchName := branch.TrackingBranch()
 		branchInfo, hasBranchInfo := branchesSnapshot.Branches.FindByRemoteName(trackingBranchName).Get()
 		if !hasBranchInfo {
-			return contributeData{}, fmt.Errorf(messages.BranchDoesntExist, trackingBranchName.String())
+			return contributeData{}, fmt.Errorf(messages.BranchDoesntExist, branch.String())
 		}
 		if branchInfo.SyncStatus == gitdomain.SyncStatusRemoteOnly {
 			branchToCheckout = Some(branch)

--- a/src/cmd/contribute.go
+++ b/src/cmd/contribute.go
@@ -146,7 +146,7 @@ func determineContributeData(args []string, repo execute.OpenRepoResult) (contri
 		trackingBranchName := branch.TrackingBranch()
 		branchInfo, hasBranchInfo := branchesSnapshot.Branches.FindByRemoteName(trackingBranchName).Get()
 		if !hasBranchInfo {
-			return contributeData{}, fmt.Errorf(messages.RemoteBranchNotFound, trackingBranchName.String())
+			return contributeData{}, fmt.Errorf(messages.BranchDoesntExist, trackingBranchName.String())
 		}
 		if branchInfo.SyncStatus == gitdomain.SyncStatusRemoteOnly {
 			branchToCheckout = Some(branch)

--- a/src/cmd/observe.go
+++ b/src/cmd/observe.go
@@ -144,7 +144,7 @@ func determineObserveData(args []string, repo execute.OpenRepoResult) (observeDa
 		trackingBranchName := branch.TrackingBranch()
 		branchInfo, hasBranchInfo := branchesSnapshot.Branches.FindByRemoteName(trackingBranchName).Get()
 		if !hasBranchInfo {
-			return observeData{}, fmt.Errorf(messages.RemoteBranchNotFound, trackingBranchName.String())
+			return observeData{}, fmt.Errorf(messages.BranchDoesntExist, branch.String())
 		}
 		if branchInfo.SyncStatus == gitdomain.SyncStatusRemoteOnly {
 			checkout = Some(branch)

--- a/src/messages/en.go
+++ b/src/messages/en.go
@@ -149,7 +149,6 @@ nd will be removed in future versions of Git Town.`
 	PushHook                       = "Push hook: %s\n"
 	PushNewBranches                = "Push new branches: %s\n"
 	RebaseProblem                  = "cannot determine rebase in progress: %w"
-	RemoteBranchNotFound           = "remote branch %q doesn't seem to exist"
 	RemoteExistsProblem            = "cannot determine if remote %q exists: %w"
 	RemotesProblem                 = "cannot determine remotes: %w"
 	RenameBranchNotInSync          = "%q is not in sync with its tracking branch, please sync the branches before renaming"

